HPCC-24523 Add filecompcopy and httpcall_xmlpost to spray class

AttilaVamos opened the pull request:
Add //class=spray tags to:
- filecompcopy.ecl
- httpcall_jsonpost.ecl
- httpcall_xmlpost.ecl

test cases to exclude them from Regression Suite on a containerized
environment.

add missing copyright information to
- httpcall_multiheader.ecl

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

HPCCSmoketest commented:
Process of PR-14048, label: HPCC-24523-impr-7.10.x starts now.
The reason of this test is: Forced to re-test.
Commit ID: d6f0b43513e0eff69f27d15152c4fa4255b7c8ad
Estimated completion time is ~0.8 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-184.ca-central-1.compute.internal

HPCCSmoketest commented:
Process of PR-14048, label: HPCC-24523-impr-7.10.x starts now.
The reason of this test is: Did not build yet.
Commit ID: d6f0b43513e0eff69f27d15152c4fa4255b7c8ad
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

AttilaVamos commented:
@jakesmith @gfortil please review.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-184.ca-central-1.compute.internal
Sha: d6f0b43513e0eff69f27d15152c4fa4255b7c8ad
Build: success
Milestone:Install hpccsystems-platform-community_7.10.11-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |144 |144 |0 |0 |0 |72 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |7 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |24 sec  (00:00:24) |
|setup (thor) |11 |11 |0 |42 sec  (00:00:42) |
|setup (roxie) |11 |11 |0 |16 sec  (00:00:16) |
|test (hthor) |883 |883 |0 |315 sec  (00:05:15) |
|test (thor) |821 |821 |0 |762 sec  (00:12:42) |
|test (roxie) |962 |962 |0 |405 sec  (00:06:45) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  752 sec (00:12:32) |  118 sec (00:01:58) |   23 sec (00:00:23) |   19 sec (00:00:19) | 1822 sec (00:30:22) |   19 sec (00:00:19) | 2762 sec (00:46:02) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: d6f0b43513e0eff69f27d15152c4fa4255b7c8ad
Build: success
Milestone:Install hpccsystems-platform-community_7.10.11-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |144 |144 |0 |0 |0 |74 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |29 sec  (00:00:29) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |20 sec  (00:00:20) |
|test (hthor) |883 |883 |0 |593 sec  (00:09:53) |
|test (thor) |821 |821 |0 |1033 sec  (00:17:13) |
|test (roxie) |962 |962 |0 |729 sec  (00:12:09) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) | 1052 sec (00:17:32) |  124 sec (00:02:04) |   22 sec (00:00:22) |   17 sec (00:00:17) | 2731 sec (00:45:31) |   19 sec (00:00:19) | 3974 sec (01:06:14) |

jakesmith commented on testing/regress/ecl/httpcall_multiheader.ecl:
```diff
@@ -1,3 +1,23 @@
+############################################################################## */
+
+//class=spray
+//class=roxieserviceaccess
```
why has 'roxieserviceaccess' been added, does this query access roxie?

AttilaVamos replied:
It was a mistake. Removed (as well as the "//class=spray" based on this isn't perform any spray operation)

jakesmith commented:
@AttilaVamos - 1 question, and related, I think PR comment is misleading:
```
Add //class=spray and //class=roxieserviceaccess tags to:

filecompcopy.ecl
httpcall_jsonpost.ecl
httpcall_multiheader.ecl
httpcall_xmlpost.ecl
```

filecompcopy isn't/doesn't have 'roxieserviceaccess',
and not sure it should have been added to httpcall_multiheader.

richardkchapman commented:
@AttilaVamos This seems to be stalled?

Should it be targeting master?

AttilaVamos commented:
@jakesmith updated, please re-review.

HPCCSmoketest commented:
Process of PR-14048, label: HPCC-24523-impr-7.10.x starts now.
The reason of this test is: Base branch changed.
Commit ID: 61424bfc17b9e06a2672c3f677af5ce5053e510e
Estimated completion time is ~0.8 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-198.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-198.ca-central-1.compute.internal
Sha: 61424bfc17b9e06a2672c3f677af5ce5053e510e
Build: success
Milestone:Install hpccsystems-platform-community_7.11.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |145 |145 |0 |0 |0 |73 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |7 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |30 sec  (00:00:30) |
|setup (thor) |11 |11 |0 |43 sec  (00:00:43) |
|setup (roxie) |11 |11 |0 |22 sec  (00:00:22) |
|test (hthor) |931 |931 |0 |352 sec  (00:05:52) |
|test (thor) |843 |843 |0 |788 sec  (00:13:08) |
|test (roxie) |1010 |1010 |0 |430 sec  (00:07:10) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  778 sec (00:12:58) |  126 sec (00:02:06) |   23 sec (00:00:23) |   19 sec (00:00:19) | 1928 sec (00:32:08) |   19 sec (00:00:19) | 2902 sec (00:48:22) |

HPCCSmoketest commented:
Process of PR-14048, label: HPCC-24523-impr-7.10.x starts now.
The reason of this test is: New commit.
Commit ID: 61424bfc17b9e06a2672c3f677af5ce5053e510e
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: 61424bfc17b9e06a2672c3f677af5ce5053e510e
Build: success
Milestone:Install hpccsystems-platform-community_7.11.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |145 |145 |0 |0 |0 |76 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |29 sec  (00:00:29) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |22 sec  (00:00:22) |
|test (hthor) |933 |933 |0 |654 sec  (00:10:54) |
|test (thor) |844 |844 |0 |1117 sec  (00:18:37) |
|test (roxie) |1012 |1012 |0 |765 sec  (00:12:45) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) | 1081 sec (00:18:01) |  126 sec (00:02:06) |   23 sec (00:00:23) |   17 sec (00:00:17) | 2919 sec (00:48:39) |   18 sec (00:00:18) | 4193 sec (01:09:53) |

HPCCSmoketest commented:
Process of PR-14048, label: HPCC-24523-impr-7.10.x starts now.
The reason of this test is: Base branch changed.
Commit ID: 61424bfc17b9e06a2672c3f677af5ce5053e510e
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: 61424bfc17b9e06a2672c3f677af5ce5053e510e
Build: success
Milestone:Install hpccsystems-platform-community_7.11.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |145 |145 |0 |0 |0 |74 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |33 sec  (00:00:33) |
|setup (thor) |11 |11 |0 |44 sec  (00:00:44) |
|setup (roxie) |11 |11 |0 |20 sec  (00:00:20) |
|test (hthor) |933 |933 |0 |671 sec  (00:11:11) |
|test (thor) |844 |844 |0 |1072 sec  (00:17:52) |
|test (roxie) |1012 |1012 |0 |765 sec  (00:12:45) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) | 1078 sec (00:17:58) |  126 sec (00:02:06) |   23 sec (00:00:23) |   17 sec (00:00:17) | 2869 sec (00:47:49) |   17 sec (00:00:17) | 4139 sec (01:08:59) |

jakesmith commented:
@AttilaVamos looks fine, but the PR commit needs changing (doesn't reflect the changes) - and the commit message not quite right (says httpcall_xmlpost.ecl changed), but hasn't.

Please change commit and PR comments and squash.

AttilaVamos committed:
HPCC-24523 Add filecompcopy and httpcall_xmlpost to spray class

Add //class=spray tags to:

- filecompcopy.ecl
- httpcall_jsonpost.ecl
- httpcall_xmlpost.ecl

test cases to exclude then from Regression Suite on a containerized
environment.

add missing copyright information to

- httpcall_multiheader.ecl

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

AttilaVamos commented:
@jakesmith updated and squashed,

HPCCSmoketest commented:
Process of PR-14048, label: HPCC-24523-impr-7.10.x starts now.
The reason of this test is: New commit.
Commit ID: b66b95ad224b28a5b4f38eaacd833c6bc7b34559
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Process of PR-14048, label: HPCC-24523-impr-7.10.x starts now.
The reason of this test is: Forced to re-test.
Commit ID: b66b95ad224b28a5b4f38eaacd833c6bc7b34559
Estimated completion time is ~0.8 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-203.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-203.ca-central-1.compute.internal
Sha: b66b95ad224b28a5b4f38eaacd833c6bc7b34559
Build: success
Milestone:Install hpccsystems-platform-community_7.11.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |145 |145 |0 |0 |0 |75 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |7 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |28 sec  (00:00:28) |
|setup (thor) |11 |11 |0 |44 sec  (00:00:44) |
|setup (roxie) |11 |11 |0 |16 sec  (00:00:16) |
|test (hthor) |933 |933 |0 |374 sec  (00:06:14) |
|test (thor) |844 |844 |0 |772 sec  (00:12:52) |
|test (roxie) |1012 |1012 |0 |430 sec  (00:07:10) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  782 sec (00:13:02) |  125 sec (00:02:05) |   23 sec (00:00:23) |   17 sec (00:00:17) | 1926 sec (00:32:06) |   18 sec (00:00:18) | 2900 sec (00:48:20) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: b66b95ad224b28a5b4f38eaacd833c6bc7b34559
Build: success
Milestone:Install hpccsystems-platform-community_7.11.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |145 |145 |0 |0 |0 |74 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |9 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |29 sec  (00:00:29) |
|setup (thor) |11 |11 |0 |44 sec  (00:00:44) |
|setup (roxie) |11 |11 |0 |22 sec  (00:00:22) |
|test (hthor) |933 |933 |0 |654 sec  (00:10:54) |
|test (thor) |844 |844 |0 |1067 sec  (00:17:47) |
|test (roxie) |1012 |1012 |0 |766 sec  (00:12:46) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) | 1095 sec (00:18:15) |  127 sec (00:02:07) |   27 sec (00:00:27) |   18 sec (00:00:18) | 2845 sec (00:47:25) |   17 sec (00:00:17) | 4138 sec (01:08:58) |

jakesmith commented:
@AttilaVamos - looks good.

jakesmith commented:
@richardkchapman - ready to merge.
